src/components/Link/Link.scss

```diff
@@ -8,15 +8,32 @@
   background: none;
   border: 0;
   font-size: inherit;
-  color: color('blue');
+  color: var(--p-interactive-action, color('blue'));
   text-decoration: none;
   cursor: pointer;
 
-  &:hover,
-  &:active {
-    color: color('blue', 'dark');
+  &:hover {
+    color: var(--p-interactive-action-hovered, color('blue', 'dark'));
     text-decoration: underline;
   }
+
+  &:focus {
+    text-decoration: var(--p-override-none, underline);
+    outline: none;
+  }
+
+  &:focus:not(:active) {
+    @include focus-ring;
+    @include plain-button-backdrop(
```

Comment:
Might be the same wrapping issue like above with adding the padding and negative margin to this state which previously did not have it.

Reply:
Looks like this is only an issue with `Link` and not `Button`. `Button` already has a base padding and negative margin so it doesn’t change the box model during active or focus. I think to achieve these focus and active styles with `Link` it will require using the `before` or `after` pseudo element.

Reply:
Or maybe we could make the base `Link` box model the same as the focus and hover styles, although that change feels riskier.

Reply:
Bummer. I was worried this might happen. I think it makes sense to cherry pick out the plain `Button` changes, and implement `Link` later on. Backwards compatibility becomes tricky, because we have to remove the existing browser focus ring on the new styles. But that becomes tricky (impossible?) to do backwards compatibly. We can specify a `outline` style, but this will be somewhat of a regression from what exists today, which is just whatever the browser has as a default.

Let's chat about this once you're online.

Reply:
I moved `Link` over to this PR: https://github.com/Shopify/polaris-react/pull/2342, so we can handle focus styles without blocking plain `Button`.